### PR TITLE
support fips-compliant build for arm

### PR DESCRIPTION
**Pull request by zhaohuabing**
Commit Message: support fips-compliant build for arm
Additional Description:  related issue: https://github.com/envoyproxy/envoy/issues/27620
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by phlax — COMMENTED** [review]
hi @zhaohuabing thanks for working on this

looking at ci and doing a quick review here im wondering how this should work, atm its failing due to trying to use non-hermetic binaries

not entirely clear how fips is expected to build wrt versions, but i think we dont want to be installing cmake/ninja etc in the host, and not sure it will work (current non-hermetic issue in ci is python afaict)

**Review by phlax — COMMENTED** [review]
@zhaohuabing i confirmed offline that the current build is not affected by the switch to the toolchain/vendored tooling
 (apologies for any confusion)

this lgtm, my one concern is that we dont have any testing for it, maybe thats ok

would be good to get a signoff from someone with better knowledge of fips

**Review by kyessenov — APPROVED** [review]
This looks fine to me. Did you try to run the ARM build and validate that fips_mode is reported in the admin? I am not sure if the CI tries this build.

Re: @phlax: FIPS mode performs a self-test so if envoy self-reports FIPS mode, it should be correct.

**Comment by phlax** [comment]
> FIPS mode performs a self-test so if envoy self-reports FIPS mode

i more meant in ci

**Comment by kyessenov** [comment]
@phlax Yeah, I'm a bit concerned about the lack of testing for compile time options for ARM. Should we create a CI pipelne similar to x86 one?

**Comment by phlax** [comment]
> Should we create a CI pipelne similar to x86 one?

the only problem is that it doesnt use a free azp host, so there is a ~cost implication - i was wondering about adding a fips build in the main release stage, but not sure

**Comment by zhaohuabing** [comment]
> I think it's OK to verify manually right now since there is no logic change. Can we get a confirmation that the ARM build reports fips_mode correctly?

@kyessenov Verified on my arm machine.

```bash
➜  echo "build --define boringssl=fips" >> .bazelrc
➜  ./ci/run_envoy_docker.sh ' ./ci/do_ci.sh bazel.release.server_only' 
➜  envoy git:(boringssl-fips-arm) ✗ lscpu|grep arc                                                          
Architecture:                    aarch64
➜  envoy git:(boringssl-fips-arm) ✗ /tmp/envoy-docker-build/envoy/arm64/source/exe/envoy/envoy --version|grep FIPS
/tmp/envoy-docker-build/envoy/arm64/source/exe/envoy/envoy  version: 2a4c291b281ecc9fad3a28868da4618eb2a6e0f2/1.27.0-dev/Modified/RELEASE/BoringSSL-FIPS
```

**Comment by wbpcode** [comment]
cc @kyessenov If this ready to merge or need a second review from other maintainers?

File: bazel/external/boringssl_fips.genrule_cmd
Comment (phlax) [inline bazel/external/boringssl_fips.genrule_cmd:91]: this is an unnecessary change, pref leave the existing formatting

File: bazel/external/boringssl_fips.genrule_cmd
Comment (phlax) [inline bazel/external/boringssl_fips.genrule_cmd:86]: i guess we will have to just go with this for now but there is no guarantee of any python in an environment

we could use the hermetic python provided by rules_python, but then im thinking rather just use `rules_foreign_cc` and avoid these problems altogether

either way, we can address the non-hermetic fips build subsequently - i appreciate this is mostly just doing for arm what is currently done for x86

**Review by phlax — COMMENTED** [review]
one small nit, and i think this should be good to land @zhaohuabing

**Review by phlax — APPROVED** [review]
lgtm, thanks @zhaohuabing

**Comment by kyessenov** [comment]
@zhaohuabing A better test is to run envoy and query server stats for `fips_mode`. Can you try that?

**Comment by zhaohuabing** [comment]
> @zhaohuabing A better test is to run envoy and query server stats for `fips_mode`. Can you try that?

Sure I can try that.

**Pull request merged**
